### PR TITLE
tests: add behavioral tests for the dbt-fal adapter

### PR DESCRIPTION
**Pull request by mederka**
For now this replaces the old dbt-fal tests.

### CODE REVIEW

**Comment by isidentical** [comment]
@mederka I think we can even delete the `test_dbt_fal` action, since it is fairly useless at this point now that we have to proper testing.

File: .github/workflows/test_adapter.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    paths:
+      - "adapter/**"
+
+  push:
+    branches: [main]
+    paths:
+      - "adapter/**"
+
+  schedule:
+    every midnight
```
Comment (isidentical) [inline .github/workflows/test_adapter.yml:15]:
```suggestion
    # every midnight
```

File: adapter/src/dbt/adapters/fal/impl.py [hunk 2]
```diff
@@ -64,18 +64,11 @@ def __new__(cls, config):
...
         original_plugin.dependencies = [db_credentials.type]
 
-        raw_credential_data = {
-            key: value
-            for key, value in db_credentials.to_dict().items()
-            if key not in db_credentials._ALIASES
-        }
-
         config.python_adapter_credentials = fal_credentials
-        config.sql_adapter_credentials = db_creds_type(**raw_credential_data)
```
Comment (chamini2) [inline adapter/src/dbt/adapters/fal/impl.py:78]: Want to do these changes in a different PR?

**Review by chamini2 — APPROVED** [review]
Just a question, but we can merge like it is! Great work.

File: adapter/src/dbt/adapters/fal/impl.py (hunk 2 above)
Comment (mederka) [reply adapter/src/dbt/adapters/fal/impl.py:78]: I think it's alright.
